src/pipeline.cc

```diff
@@ -217,7 +217,7 @@ class PipelineWorker : public Napi::AsyncWorker {
           ->set("access", baton->input->access)
           ->set("scale", scale)
           ->set("fail_on", baton->input->failOn);
-        if (inputImageType == sharp::ImageType::WEBP) {
+        if (scale < 1.0 && inputImageType == sharp::ImageType::WEBP) {
```

Comment:
This causes a small memory leak for the dynamic-allocated `*option` when `scale > 1.0`, since it's never `delete`'d. How about doing this?
```diff
--- a/src/pipeline.cc
+++ b/src/pipeline.cc
@@ -188,8 +188,10 @@ class PipelineWorker : public Napi::AsyncWorker {
           if (jpegShrinkOnLoad > 1 && static_cast<int>(shrink) == jpegShrinkOnLoad) {
             jpegShrinkOnLoad /= 2;
           }
-        } else if (inputImageType == sharp::ImageType::WEBP ||
-                   inputImageType == sharp::ImageType::SVG ||
+        } else if (inputImageType == sharp::ImageType::WEBP && shrink > 1.0) {
+          // Avoid upscaling via libwebp
+          scale = 1.0 / shrink;
+        } else if (inputImageType == sharp::ImageType::SVG ||
                    inputImageType == sharp::ImageType::PDF) {
           scale = 1.0 / shrink;
         }
```